romancal/dark_current/dark_current_step.py

```diff
@@ -42,7 +42,8 @@ def process(self, input):
 
             # Do the dark correction
             out_model = input_model
-            out_model.data -= dark_model.data
+            nresultants = len(input_model.meta.exposure["read_pattern"])
+            out_model.data -= dark_model.data[:nresultants]
```

Comment:
```suggestion
            n_resultants = len(input_model.meta.exposure["read_pattern"])
            out_model.data -= dark_model.data[:n_resultants]
```
I find this more readable